### ISSUE TITLE
import jarray ModuleNotFoundError: No module named 'jarray'

### ISSUE THREAD

**Issue by HermanTam** [NationalSecurityAgency/ghidra#6027]
currentProgram.getMemory().getBytes(address_to_read, bytes_read) does not populate my byte array. The byte array remains the same before and after calling the method. 
I found https://github.com/NationalSecurityAgency/ghidra/issues/858, it appears I have to use jarray. However, when trying to import jarray, it returns "import jarray
ModuleNotFoundError: No module named 'jarray'". I cannot find the jarray package to install. 

import jarray
bytes = jarray.array(bytes_read, "b")

**My attempts without using jarray**
My code attempts to read bb.size bytes from a specified address into the bytes_read array, but bytes_read remains the same before and after calling getBytes method. 
address_to_read = bb.getMinAddress()

# Initialize a byte array to store the read bytes
bytes_read = bytearray(bb.size)
print("bytes_read before: ", bytes_read)
currentProgram = getCurrentProgram()

# Attempt to read bytes from the specified address into the byte array
num_bytes_read = currentProgram.getMemory().getBytes(address_to_read, bytes_read)

print("bytes_read after: ", bytes_read)
print("num_bytes_read: ", num_bytes_read)

bytes_read before: bytearray(b'\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00')
bytes_read after: bytearray(b'\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00')
num_bytes_read: 12

**Expected behavior**

**Screenshots**
I've tried the following to initialize bytes_read:

- bytes_read = array.array('b', bytes_read)
- bytes_read = array.array('b', repr(bytes(bytes_read))[2:-1])
- bytes_read = array.array('b', bytes_read.decode('utf-8'))
- bytes_read = array.array('b', bytes_read.decode())
- bytes_read = list()*bb.size

- num_bytes_read = currentProgram.getMemory().getBytes(address_to_read, bytes_read, 0, bb.size)

None of them populate bytes_read.

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
 - OS: Windows 11 
 - Java Version: 21.0.1
 - Ghidra Version: 10.4
 - Ghidra Origin: official GitHub distro
 - Ghidrathon Python 3 

Any guidance is highly appreciated.

**Comment by HermanTam** [comment]
I think I will iterate through the address set and extract the bytes one by one with the getByte method as a workaround for now before this is updated. 

I also found that this method 'public byte getByte​([Address] addr) throws [MemoryAccessException]' returns an int instead of byte as in the documentation https://ghidra.re/ghidra_docs/api/ghidra/program/database/mem/MemoryBlockDB.html#getByte(ghidra.program.model.address.Address)

**Comment by d-millar** [comment]
Are you accessing the Ghidra API through the Jython interface or Python 3 via Ghidrathon?

**Comment by HermanTam** [comment]
Python 3 via Ghidrathon
My current workaround is using MemoryByteIterator(currentProgram().getMemory(), block_address_set).next().

**Comment by d-millar** [comment]
So, the jarray solution only applies to Jython, i.e. using the default Ghidra scripting interface.  Just to be clear, "bytearray" failed as well?  I.e. you tried that before resorting to the solution in #858, yes?

**Comment by HermanTam** [comment]
Yes, all options mentioned in the post failed including bytes_read = bytearray(bb.size).

**Comment by d-millar** [comment]
OK, borrowing from the Ghidrathon documentation, I tried the following successfully:

currentAddress = currentSelection().getMinAddress()
code = bytes(map(lambda b: b & 0xff, getBytes(currentAddress, 10)))

"code" correctly receives the byte values from my selection.
Does this work for you?
